mantidproject/mantid
# Create Muon Nexus V2 loading algorithm

**Description of work.**
This PR creates a new Muon Nexus loading Algorithm to load the Muon v2 files.

Currently we only have single period files and therefore the single period and Multi period loading has been separated though the use of the strategy pattern. Once available, we will implement the MultiPeriodLoadMuonStrategy.

This new loader works with the Load algorithm but not with the existing LoadMuonNexus algorithm, this means it has not yet been implemented in the GUI. The reason for this is due to different integer types used in LoadISISNexus and LoadMuonNexus, which means they do not adhere to the same interface. In this PR, I began changing the interfaces to use the same integer types, but it soon became apparent this change was outside the scope of this PR due to the number of places where integer types had to be changed (~200-300 lines). Therefore, it will be dealt with in a separate issue #28695.

Lastly, this PR also forces the LoadISISNexus call to be single threaded, this was favored as it was found that the multi-threaded call was slower (sometimes considerably) than the single threaded call, due to the small workspaces loaded vs the overhead of setting up new threads.





**To test:**
Turn the archive on and load the workspaces using the original and new algorithms

1. LoadMuonNexus(filename="EMU00103640.nxs", OutputWorkspace="wsv1") # v1 file
2. LoadMuonNexusV2(filename="EMU00103640.nxs_v2",OutputWorkspace="wsv2") # v2 file

The workspaces should be identical, although there are a few differences the logs - namely LoadMuonNexusV2 is missing the entries:
- durunits
- dur_secs
- ICPEVENT 
- ISIS beam

But ICPEvent is available in lower case form icp_event and the duration can be found from the   run_start and run_end entries.

Build the documentation and make sure it makes sense
 


Fixes #27701. 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
